pyccel/codegen/printing/cwrappercode.py

```diff
@@ -0,0 +1,236 @@
+# coding: utf-8
+# pylint: disable=R0201
+
+from pyccel.codegen.printing.ccode import CCodePrinter
+
+from pyccel.ast.builtins import Bool
+
+from pyccel.ast.core import Variable, ValuedVariable, Assign, AliasAssign, FunctionDef
+from pyccel.ast.core import If, Nil, Return, FunctionCall, PyccelNot
+from pyccel.ast.core import create_incremented_string
+
+from pyccel.ast.datatypes import NativeInteger, NativeBool
+
+from pyccel.ast.cwrapper import PyccelPyObject, PyArg_ParseTupleNode, PyBuildValueNode
+from pyccel.ast.cwrapper import PyArgKeywords
+
+from pyccel.ast.type_inference import str_dtype
+
+from pyccel.errors.errors import Errors
+from pyccel.errors.messages import *
+
+errors = Errors()
+
+__all__ = ["CWrapperCodePrinter", "cwrappercode"]
+
+class CWrapperCodePrinter(CCodePrinter):
+    def __init__(self, parser, settings={}):
+        CCodePrinter.__init__(self, parser,settings)
+
+    def get_new_name(self, used_names, requested_name):
+        if requested_name not in used_names:
+            used_names.add(requested_name)
+            return requested_name
+        else:
+            return create_incremented_string(used_names, prefix=requested_name)
+
+    def get_PyArgParseType(self, dtype):
+        #TODO: Depends on type, rank, etc
+        if dtype is NativeBool():
+            return NativeInteger(), lambda arg, tmp: Assign(arg, Bool(tmp))
```

Comment:
This needs to be a NativeInteger of precision 4. Should this function receive a Variable and return a Variable and a function instead of receiving a dtype and returning a dtype and a function?

Reply:
collecting variable and function is a good idea  but we need to pass used_name list and the  arg name too in the function args

Reply:
or we can keep the code as is it and return a third paramete    precision  with dtype and function .

Reply:
The arg name should be a property of the variable

Reply:
I think it's better to return a Variable. Who knows what other information we may need to add later (e.g. I think optional variables will be pointers) and we don't want to add too many return arguments. Plus the less we do in the for loop the more chance we have of reducing that code to 1 line